File: administrator/src/main/java/app/downloadaccess/administrator/network/RetrofitService.java
package app.downloadaccess.administrator.network;

import com.google.gson.JsonObject;

import java.util.Map;

import app.downloadaccess.resources.models.Place;
import app.downloadaccess.resources.models.Slot;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import retrofit2.http.QueryMap;

public interface RetrofitService {

    @GET("/api/user")
    Call<JsonObject> getUserId();

    @GET("/api/place-types")
    Call<JsonObject> getPlaceTypes();

    @GET("/api/place/{visitorId}")
    Call<JsonObject> getAllPlaces(@Path("visitorId") String visitorId, @QueryMap Map<String, Object> map);

    @GET("/api/place/{placeId}/approve/{approvedStatus}")
    Call<JsonObject> setApproved(@Path("placeId") String placeId, @Path("approvedStatus") boolean approvedStatus);

    @GET("/api/place/{visitorId}/{placeId}")
    Call<JsonObject> getSlotsPlace(@Path("visitorId") String visitorId, @Path("placeId") String placeId);

    @GET("/api/user/{visitorId}/visits")
    Call<JsonObject> getUserVisits(@Path("visitorId") String visitorId);

    @DELETE("/api/user/{visitorId}/visit/{slotId}")
    Call<JsonObject> deleteVisit(@Path("visitorId") String visitorId, @Path("slotId") String slotId);

    @POST("/api/place/{visitorId}")
    Call<Void> addPlace(@Body Place place, @Path("visitorId") String visitorId);

    @PUT("/api/place/{visitorId}/{placeId}")
    Call<Void> editPlace(@Body Place place, @Path("visitorId") String visitorId, @Path("placeId") String placeId);

    @POST("/api/slot/{placeId}")
    Call<Void> addSlot(@Path("placeId") String placeId, @Body Slot slot);

}

File: resources/src/main/java/app/downloadaccess/resources/network/RetrofitService.kt
package app.downloadaccess.resources.network

import app.downloadaccess.resources.models.Place
import app.downloadaccess.resources.models.Slot
import com.google.gson.JsonObject
import retrofit2.Call
import retrofit2.http.*

interface RetrofitService {
    @GET("generate-user-id")
    fun getUserId(@Header("Authorization") token: String?): Call<JsonObject>

    @GET("get-place-types")
    fun getPlaceTypes(@Header("Authorization") token: String?): Call<JsonObject>

    @POST("approve-place/{placeId}/{approvedStatus}")
    fun setApproved(
        @Header("Authorization") token: String?,
        @Path("placeId") placeId: String?,
        @Path("approvedStatus") approvedStatus: Boolean
    ): Call<JsonObject?>?

    @POST("get-places/{userId}")
    fun getAllPlaces(
        @Header("Authorization") token: String?,
        @Path("userId") placeId: String?,
        @Body body: HashMap<String?, Any?>?
    ): Call<JsonObject>

    @GET("get-place-slots/{userId}/{placeId}")
    fun getSlotsPlace(
        @Header("Authorization") token: String?,
        @Path("userId") userId: String?,
        @Path("placeId") placeId: String?
    ): Call<JsonObject>

    @GET("get-bookings/{userId}")
    fun getUserVisits(
        @Header("Authorization") token: String?,
        @Path("userId") userId: String?
    ): Call<JsonObject>

    @GET("delete-booking/{userId}/{slotId}")
    fun deleteVisit(
        @Header("Authorization") token: String?,
        @Path("userId") userId: String?,
        @Path("slotId") slotId: String?
    ): Call<JsonObject>

    @DELETE("delete-slot/{slotId}/{userId}")
    fun deleteSlot(
        @Header("Authorization") token: String?,
        @Path("slotId") slotId: String?,
        @Path("userId") userId: String?
    ): Call<JsonObject>

    @POST("add-place")
    fun addPlace(
        @Header("Authorization") token: String?,
        @Body place: Place?
    ): Call<Void>

    @PUT("update-place/{placeId}")
    fun editPlace(
        @Header("Authorization") token: String?,
        @Body place: Place?,
        @Path("placeId") placeId: String?
    ): Call<Void>

    @POST("add-slot/{placeId}")
    fun addSlot(
        @Header("Authorization") token: String?,
        @Path("placeId") placeId: String?,
        @Body slot: Slot?
    ): Call<Void>

    @GET("change-favourite/{userId}/{placeId}")
    fun addRemoveFavourite(
        @Header("Authorization") token: String?,
        @Path("userId") userId: String?,
        @Path("placeId") placeId: String?
    ): Call<JsonObject?>?

    @POST("visit")
    fun planVisit(
        @Header("Authorization") token: String?,
        @Body body: HashMap<String?, Any?>?
    ): Call<JsonObject?>?
}